cassh

```diff
@@ -110,44 +103,49 @@ class CASSH(object):
         self.auth = user_metadata['auth']
         self.realname = user_metadata['realname']
 
-
-    def auth_url(self, prefix=None):
+    def get_data(self, prefix=None):
         """
-        Return a xxx=xxx to put at the end of a GET request.
+        Return data for a POST request.
         """
+        data = {}
         passwd_message = 'Please type your LDAP password (user=%s): ' % self.realname
         if self.auth == 'ldap':
-            auth = {'realname': self.realname, 'password': getpass(passwd_message)}
-            if prefix is not None:
-                auth.update(prefix)
-            return '?%s' % urlencode(auth)
-        else:
-            if prefix is None:
-                return ''
-            return '?%s' % urlencode(prefix)
+            data.update({'realname': self.realname, 'password': getpass(passwd_message)})
+        if prefix is not None:
+            data.update(prefix)
+        return data
 
     def admin(self, username, action, set_value=None):
         """
         Admin CLI
         """
+        payload = self.get_data()
         try:
             if action == 'revoke':
```

Comment:
FAR (For Another Review): a switch/case would be more relevant instead of multiple if/elif conditions

Reply:
Switch does not exists in python ....

Reply:
You're goddamn right ! *facepalm*